src/overlays/actors/ovl_Boss_Goma/z_boss_goma.c

```diff
@@ -21,115 +82,1778 @@ const ActorInit Boss_Goma_InitVars = {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005F28, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06005F28.genericHeader),
+                         2, -2.0f);
+    this->actionFunc = func_809190F8;
+    this->unk_21C = SkelAnime_GetFrameCount(&D_06005F28.genericHeader);
+}
+
+void func_809165C0(BossGoma* this) {
+    f32 tmp;
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_060058C0, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060058C0.genericHeader),
+                         2, -2.0f);
+    tmp = SkelAnime_GetFrameCount(&D_060058C0.genericHeader);
+    this->actionFunc = func_80919054;
+    this->unk_21C = tmp;
+    this->unk_21C = SkelAnime_GetFrameCount(&D_060058C0.genericHeader);
+}
+
+void func_80916668(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600CCD8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0600CCD8.genericHeader),
+                         0, -2.0f);
+    this->actionFunc = func_80919150;
+}
+
+void func_809166D8(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_0600C468, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_0600C468.genericHeader),
+                         2, -10.0f);
+    this->actionFunc = func_80918C08;
+}
+
+void func_8091674C(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000AE8, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06000AE8.genericHeader),
+                         0, -10.0f);
+    this->actionFunc = func_80918D10;
+    this->unk_1D2 = 0;
+}
+
+void func_809167C0(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06004318, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06004318.genericHeader),
+                         2, -10.0f);
+    this->actionFunc = func_80918D6C;
+    this->unk_1D0 = 0;
+    this->unk_1D2 = 0;
+}
+
+void func_8091683C(BossGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06006E60, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06006E60.genericHeader),
+                         2, -2.0f);
+    this->actionFunc = func_80918FBC;
+}
+
+void func_809168B0(BossGoma* this, GlobalContext* globalCtx, f32 arg2, f32 arg3, s16 arg4) {
+    static Vec3f D_8091B2B4 = { 0.0f, 0.0f, 0.0f };
+    static Vec3f D_8091B2C0 = { 0.0f, -0.5f, 0.0f };
+    static Vec3f D_8091B2CC = { -150.0f, 0.0f, -350.0f };
+
+    s16 i;
+    Vec3f* tmp = NULL;
+    Vec3f a;
+    Vec3f b;
+    Vec3f c;
+
+    D_8091B2CC.z += arg2;
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothScaleMaxF(&this->actor.speedXZ, arg3, 0.5f, 2.0f);
+    if (arg4) {
+        Math_SmoothScaleMaxS(&this->actor.posRot.rot.y, Math_Vec3f_Yaw(&this->actor.posRot.pos, &D_8091B2CC) + 0x8000,
+                             3, 0x3E8);
+    }
+
+    if (func_800A56C8(&this->skelAnime, 9.0f)) {
+        tmp = &this->unk_278;
+    } else if (func_800A56C8(&this->skelAnime, 1.0f)) {
+        tmp = &this->unk_284;
+    }
+
+    if (tmp != NULL) {
+        for (i = 0; i < 5; i++) {
+            a = D_8091B2B4;
+            b = D_8091B2C0;
+            c.x = Math_Rand_CenteredFloat(70.0f) + tmp->x;
```

Comment:
like we discussed in discord, declaring in the loop is cleaner

Reply:
Can't be done there due to order of accesses.